frank-doc-frontend/src/app/pipes/as-text.pipe.ts

```diff
@@ -9,33 +9,39 @@ export class AsTextPipe implements PipeTransform {
     if (!value) return "";
     value = value.replaceAll('\\"', '"');
     value = value.replace(/<[^>]*>?/gm, '');
-    value = value.replace(/(?:{@link\s(.*?)})/g, function (match, captureGroup) {
-      // {@link PipeLineSession pipeLineSession}
-      // {@link IPipe#configure()}
-      // {@link #doPipe(Message, PipeLineSession) doPipe}
-      let referencedElement = captureGroup;
-      const hash = captureGroup.indexOf("#");
-      if (hash > -1) {
-        referencedElement = captureGroup.split("#")[0];
+    value = value.replace(/(?:{@link\s(.*?)})/g, (match, captureGroup: string) => {
+      // {@link PipeLineSession pipeLineSession} -> 'PipeLineSession pipeLineSession'
+      // {@link IPipe#configure()} -> 'IPipe#configure()'
+      // {@link #doPipe(Message, PipeLineSession) doPipe} -> '#doPipe(Message, PipeLineSession) doPipe'
+      const hashPos = captureGroup.indexOf("#"),
+        isMethod = hashPos > -1,
+        element = isMethod ? captureGroup.split("#")[0] : captureGroup;
 
-        if (referencedElement == '') { //if there is no element ref then it's a method
-          const method = captureGroup.slice(hash);
-          const nameOrAlias = method.split(") ");
-          if (nameOrAlias.length == 2) {
-            return nameOrAlias[1]; //If it's an alias
-          }
-          return method.slice(1, method.indexOf("("));
-        }
+      if (element == '') { //if there is no element ref then it's a method
+        return this.transformInternalMethod(captureGroup, hashPos);
       }
-      const captures = referencedElement.split(" "); //first part is the ClassName, 2nd part the written name
-      let name = captures[captures.length - 1];
-      if (hash > -1) {
-        const method = captureGroup.split("#")[1];
-        name = name + "." + (method.slice(method.indexOf(") ") + 1)).trim();
-      }
-      return name;
+
+      const elementParts = element.split(" "); //first part is the class name, 2nd part the written name
+      return this.name(elementParts, isMethod, captureGroup);
     });
     return value;
   }
 
+  transformInternalMethod(captureGroup: string, hashPos: number) {
+    const methodName = captureGroup.slice(hashPos),
+      methodLabelSplit = methodName.split(" ");
+
+    if (methodLabelSplit.length == 2) return methodLabelSplit[1]; //return method label
+    return methodName.slice(1, methodName.indexOf("("));
+  }
+
+  name(elementParts: string[], isMethod: boolean, captureGroup: string) {
```

Comment:
Oh ik dacht dat dat andere de naam was. Maar als ik het zo lees is het andersom?
Deze bewerkingen begrijp ik niet zo. Maar ik weet ook niet goed wat de input is, dus misschien verklaart dat wel.